payments-core/src/main/java/com/stripe/android/IntentConfirmationInterceptor.kt

```diff
@@ -161,7 +166,10 @@ class DefaultIntentConfirmationInterceptor @Inject constructor(
                 createConfirmStep(result.clientSecret, shippingValues, paymentMethod)
             }
             is CreateIntentCallback.Result.Failure -> {
-                IntentConfirmationInterceptor.NextStep.Fail(result.errorMessage)
+                IntentConfirmationInterceptor.NextStep.Fail(
+                    cause = result.cause,
+                    message = result.cause.message ?: genericErrorMessage,
```

Comment:
I don't think we should put exception error messages into the UI. Should this always be `genericErrorMessage`

Reply:
These are exceptions returned by the merchant in the `createIntentCallback`. Our thinking was that they might relay some domain-specific information, e.g. `You can’t buy the cookies because you’ve run out of cookie points` or something like that. Do you think we should not allow that?

Reply:
We should optionally allow them to pass in a user visible message. If it's null, we can just show the generic message.

Reply:
Meaning that `CreateIntentCallback.Result.Failure` would have both an `error` and a `message`?

Reply:
That would be my vote.

`class Failure (internal val error: Exception, internal val message: String? = null)`

Reply:
Done

Reply:
Should we call it `displayMessage` then to make it clear how it’ll be used?

Reply:
I think so.

Reply:
Done